src/net_processing.cpp

```diff
@@ -4400,7 +4414,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                 // process from some other peer.  We do this after calling
                 // ProcessNewBlock so that a malleated cmpctblock announcement
                 // can't be used to interfere with block relay.
-                RemoveBlockRequest(pblock->GetHash());
+                RemoveBlockRequest(pblock->GetHash(), std::nullopt);
```

Comment:
Is this block required anymore now that `ProcessBlock` will call it for us?

Reply:
Can't we have a race condition where the block is not `new`? In any case this function is extremely fast if called twice.